go/interpreter/lfvm/memory_test.go

```diff
@@ -47,3 +54,230 @@ func TestExpansionCosts(t *testing.T) {
+				copy(m.store[24:], testValue)
+				ctxt.gas = 100
+			},
+			offset:         0,
+			expectedTarget: *uint256.NewInt(0).SetBytes8(testValue),
+			expectedError:  nil},
+		"not enough gas": {
+			setup: func(m *Memory, ctxt *context) {
+				m.store = make([]byte, 32)
+				copy(m.store[24:], testValue)
+				ctxt.gas = 6
+			},
+			offset:         64,
+			expectedTarget: *uint256.NewInt(0),
+			expectedError:  errOutOfGas},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			target := uint256.NewInt(0)
+			test.setup(m, &ctxt)
+
+			err := m.CopyWord(test.offset, target, &ctxt)
+			if err != test.expectedError {
+				t.Errorf("unexpected error, want: %v, got: %v", test.expectedError, err)
+			}
+			if target.Cmp(&test.expectedTarget) != 0 {
+				t.Errorf("unexpected target value, want: %v, got: %v", test.expectedTarget, target)
+			}
+		})
+	}
+}
+
+func TestMemory_SetByte(t *testing.T) {
+
+	tests := map[string]struct {
+		value    byte
+		offset   uint64
+		gas      tosca.Gas
+		expected error
+	}{
+		"regular":        {value: 0x12, offset: 0, gas: 100, expected: nil},
+		"not enough gas": {value: 0x12, offset: 64, gas: 0, expected: errOutOfGas},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			ctxt.gas = test.gas
+
+			err := m.SetByte(test.offset, test.value, &ctxt)
+			if err != test.expected {
+				t.Errorf("unexpected error, want: %v, got: %v", test.expected, err)
+			}
+			if err == nil && m.store[test.offset] != test.value {
+				t.Errorf("unexpected value, want: %v, got: %v", test.value, m.store[test.offset])
+			}
+		})
+	}
+}
+
+func TestSetWordProperlyReportsNotEnoughGas(t *testing.T) {
+	ctxt := getEmptyContext()
+	m := NewMemory()
+	ctxt.gas = 0
+	err := m.SetWord(0, uint256.NewInt(0), &ctxt)
+	if err != errOutOfGas {
+		t.Errorf("unexpected error, want: %v, got: %v", errOutOfGas, err)
+	}
+}
+
+func TestMemory_Set(t *testing.T) {
+
+	tests := map[string]struct {
+		size     uint64
+		offset   uint64
+		expected error
+	}{
+		"regular":         {size: 8, offset: 0, expected: nil},
+		"size overflow":   {size: math.MaxUint64, offset: 1, expected: errGasUintOverflow},
+		"offset overflow": {size: 32, offset: math.MaxUint64, expected: errGasUintOverflow},
+		"not enough memory": {size: 32, offset: 32,
+			expected: fmt.Errorf("memory too small, size %d, attempted to write %d bytes at %d", 8, 32, 32)},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			m.store = make([]byte, 8)
+			ctxt.gas = 100
+
+			err := m.Set(test.offset, test.size, []byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef})
+
+			if err == nil {
+				if test.expected != nil {
+					t.Errorf("expected error %v, got nil", test.expected)
+				}
+			} else if !strings.Contains(err.Error(), test.expected.Error()) {
```

Comment:
currently, `Set` returns an error that is instantiated as needed, so `errors.Is` does not work, and I need to compare string. Also not sure I could make the error into a `const` (this would enable using `errors.is` ), because the error takes a couple of parameters.

Reply:
I've added an error generating function, that allows the use of `errors.Is`, we should make the usage of such a functions a project wide practice, since we don't need to compare strings anymore.

Reply:
I agree